battle-engine.js

```diff
@@ -492,6 +492,56 @@ BattlePokemon = (function () {
 		}
 		return null;
 	};
+	BattlePokemon.prototype.getMoveTargets = function (move, target) {
+		move = this.battle.getMove(move);
+		var targets = [];
+		switch (move.target) {
+		case 'all':
+		case 'foeSide':
+		case 'allySide':
+		case 'allyTeam':
+			if (move.target.substr(0, 3) !== 'foe') {
```

Comment:
I thought neither. I thought "first X characters".

Perhaps I should just use `String.startsWith`. It might be faster performance-wise anyways since you don't need to build a new string